apps/api/src/middleware/active-comp-filter.middleware.ts

```diff
@@ -0,0 +1,73 @@
+import { eq } from "drizzle-orm";
+import { NextFunction, Request, Response } from "express";
+
+import { config } from "@/config/index.js";
+import { db } from "@/database/db.js";
+import { competitions } from "@/database/schema/core/defs.js";
+import { middlewareLogger } from "@/lib/logger.js";
+import { ApiError } from "@/middleware/errorHandler.js";
+import { COMPETITION_STATUS } from "@/types/index.js";
+
+/**
+ * Active Competition Filter Middleware
+ *
+ * This middleware checks if there is an active competition. If no active competition
+ * is found, it responds with a 403 Forbidden error.
+ */
+export const activeCompMiddleware = function () {
+  return async function (req: Request, res: Response, next: NextFunction) {
+    try {
+      const activeCompetition = await getActiveComp();
+      if (!activeCompetition) {
+        middlewareLogger.debug("No active competition found");
+        throw new ApiError(403, "No active competition");
+      }
+
+      middlewareLogger.debug(
+        `Active competition found: ${activeCompetition.id}`,
+      );
+
+      // Set competition ID in request for downstream use
+      req.competitionId = activeCompetition.id;
+
+      next();
+    } catch (error) {
+      middlewareLogger.error(`Error checking active competition:`, error);
+      next(error);
+    }
+  };
+};
+
+// Cache for active competition check
+let cachedActiveCompetition: { id: string } | null = null;
+let lastQueryTime = 0;
+const CACHE_ACTIVE_COMP_TTL_MS = config.cache.activeCompetitionTtlMs;
```

Comment:
Since this is a very short (default 3 seconds) lived cache and a relatively low overhead query, seems fine to just have a per instance in-memory cache that can't be manually reset.  
If we revisit caching in a different layer of the app, we can potentially re-think this.